lib/airrecord/faraday_rate_limiter.rb

```diff
@@ -0,0 +1,60 @@
+require 'thread'
+
+module Airrecord
+  class FaradayRateLimiter < Faraday::Middleware
+    class << self
+      attr_accessor :requests
+    end
+
+    def initialize(app, requests_per_second: nil, sleeper: nil)
+      super(app)
+      @rps = requests_per_second
+      @sleeper = sleeper || ->(seconds) { sleep(seconds) }
+      @mutex = Mutex.new
+      clear
+    end
+
+    def call(env)
+      @mutex.synchronize do
```

Comment:
This probably seems a bit heavy-handed—however, I'd rather just have a simple, global mutex when I can because it's dead-simple to reason about. You're not going to get more throughput with threads anyway, but I don't see any reason why you wouldn't be able to use threads for Airrecord.